model_compression_toolkit/trainable_infrastructure/pytorch/base_pytorch_quantizer.py

```diff
@@ -53,6 +43,12 @@ def get_trainable_variables(self, group: VariableGroup) -> List[torch.Tensor]:
                 quantizer_parameter, parameter_group = parameter_dict[VAR], parameter_dict[GROUP]
                 if quantizer_parameter.requires_grad and parameter_group == group:
                     quantizer_trainable.append(quantizer_parameter)
+
+            # sanity check to catch inconsistent initialization
+            if self.freeze_quant_params and group == VariableGroup.QPARAMS:
+                assert not quantizer_trainable, ('Found trainable quantization params despite self.freeze_quant_params=True. '
```

Comment:
I won't argue about this if you insist, but:
a. I don't view assertion as part of error handling. The difference is subtle and not always clear, but here it just serves as a sanity check to alert developers that they are trying to do something that doesn't make sense. I definitely wouldn't use it to validate user input for example. I can change it to RuntimeError or something if you prefer.

b. What is the value of logging error message immediately followed by an exception with the same error message? Noone is going to scroll up above the exception to see the error message and even if someone does, it doesn't add any info. Error logging has value when you catch the exception and just log a nice clear message, or at least add some information.

Reply:
All true, but currently the entire code is working with this convention, so I prefer to keep working with it instead of mixing different behaviors in different places of the code.
Hopefully, we will address this in the near future and change this throughout the code.

Reply:
will be fixes in #1197